automation/taskcluster/decision_task_nightly.py

```diff
@@ -3,95 +3,154 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 """
-Decision task for releases
+Decision task for nightly releases.
 """
 
 from __future__ import print_function
-import datetime
+
+import argparse
 import json
+import lib.tasks
 import os
 import taskcluster
 
-import lib.tasks
-
 TASK_ID = os.environ.get('TASK_ID')
+SCHEDULER_ID = os.environ.get('SCHEDULER_ID')
+GITHUB_HTTP_REPOSITORY = os.environ.get('MOBILE_HEAD_REPOSITORY')
 HEAD_REV = os.environ.get('MOBILE_HEAD_REV')
 
-def generate_build_task():
-    created = datetime.datetime.now()
-    expires = taskcluster.fromNow('1 year')
-    deadline = taskcluster.fromNow('1 day')
-
-    command = "automation/taskcluster/actions/nightly.sh"
-
-    return {
-        "workerType": 'gecko-focus',
-        "taskGroupId": TASK_ID,
-        "expires": taskcluster.stringDate(expires),
-        "retries": 5,
-        "created": taskcluster.stringDate(created),
-        "tags": {},
-        "priority": "lowest",
-        "schedulerId": "focus-nightly-sched",
-        "deadline": taskcluster.stringDate(deadline),
-        "dependencies": [ TASK_ID ],
-        "routes": [
-            "index.project.mobile.reference-browser.nightly.latest"
-        ],
-        "scopes": [
-            "queue:route:index.project.mobile.reference-browser.nightly.*",
-            "secrets:get:project/mobile/reference-browser/preview-key-store",
+BUILDER = lib.tasks.TaskBuilder(
+    task_id=TASK_ID,
+    owner="skaspari@mozilla.com",
+    source='{}/raw/{}/.taskcluster.yml'.format(GITHUB_HTTP_REPOSITORY, HEAD_REV),
+    scheduler_id=SCHEDULER_ID
+)
+
+
+def generate_build_task(apks):
+    artifacts = {}
+    for apk in apks:
+        artifact = {
+            "type": 'file',
+            "path": "/build/reference-browser/{}".format(apk),
+            "expires": taskcluster.stringDate(taskcluster.fromNow('1 year'))
+        }
+        artifacts["public/{}".format(os.path.basename(apk))] = artifact
```

Comment:
Nit: After list-comprehension, there's dict-comprehension :smiley: (not blocking)

Reply:
Aha, nice!